### PR TITLE
Move sbt-plugin-test to scripted

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
I'm not 100% sure what to do about all the `project/` directories. Do you think a symlink would be acceptable?

**Comment by sjrd** [comment]
Symlinks are a big no no on Windows, so copy paste is unfortunately preferable.

File: sbt-plugin/src/sbt-test/testing/multi-framework/build.sbt [hunk 1]
```diff
@@ -0,0 +1,62 @@
+lazy val versionSettings = Def.settings(
+    version := scalaJSVersion,
+    scalaVersion := "2.12.10"
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/testing/multi-framework/build.sbt]: Consider putting those `inThisBuild(...)` instead.

File: sbt-plugin/src/sbt-test/testing/multi-framework/project/build.sbt [hunk 2]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % sys.props("plugin.version"))
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/testing/multi-framework/project/build.sbt]: Here you've inlined `sys.props(...)` but in other tests you used a separate `val scalaJSVersion`. Consider making this uniform, one way or the other.

File: sbt-plugin/src/sbt-test/testing/multi-framework/test [hunk 3]
```diff
@@ -0,0 +1,5 @@
+> root/test
+> multiTestJS/testHtml
+> set scalaJSStage in Global := FullOptStage
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/testing/multi-framework/test]: Does this actually *do* anything on its own? Are multiple `>` commands executed within the same session? My impression is that each `>` command reloads sbt, which means that any `set` command on its own is immediately discarded.

File: sbt-plugin/src/sbt-test/linker/non-existent-classpath/project/build.sbt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+val scalaJSVersion = sys.props("plugin.version")
+
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % scalaJSVersion)
+
+libraryDependencies += "org.scala-js" %% "scalajs-linker" % scalaJSVersion
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/linker/non-existent-classpath/project/build.sbt]: This seems unused.

File: sbt-plugin/src/sbt-test/cross-version/2.11/test [hunk 5]
```diff
@@ -0,0 +1,8 @@
+> run
+> test
+> testHtml
+> clean
+> set scalaJSStage in Global := FullOptStage
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/cross-version/2.11/test]: Same concern about the lonely `set` command.

File: sbt-plugin-test/build.sbt [hunk 6]
```diff
@@ -33,20 +28,6 @@ lazy val noDOM = project.settings(baseSettings: _*).
   settings(
     name := "Scala.js sbt test w/o DOM",
     scalaJSUseMainModuleInitializer := true,
-    scalaJSModuleInitializers +=
-      ModuleInitializer.mainMethod("sbttest.noDOM.TestApp", "foo"),
```
Comment (sjrd) [inline sbt-plugin-test/build.sbt]: This change should be accompanied by removing `def foo()` in `TestApp.scala`.

File: sbt-plugin/src/sbt-test/cross-version/2.13/src/main/scala/sbttest/TestApp.scala [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+object TestApp {
+
+  def main(args: Array[String]): Unit = {
+    println(Lib.foo("Hello World"))
+    println(Lib.sq(10))
+  }
+
+  // Used as a manual module initializer.
+  def foo(): Unit = ()
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/cross-version/2.13/src/main/scala/sbttest/TestApp.scala]: This is unused. (it should also not appear in the 2.11 `TestApp.scala`, but that should be removed in a previous commit already (see my other comment on the topic).

File: sbt-plugin/src/sbt-test/cross-version/2.13/test [hunk 8]
```diff
@@ -0,0 +1,8 @@
+> run
+> test
+> testHtml
+> clean
+> set scalaJSStage in Global := FullOptStage
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/cross-version/2.13/test]: Again the lonely `set`.

File: sbt-plugin/src/sbt-test/cross-version/2.13/src/main/scala/sbttest/Lib.scala [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package sbttest.noDOM
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/cross-version/2.13/src/main/scala/sbttest/Lib.scala]: Perhaps use the opportunity to rename this package. The test doesn't have anything to do with DOM or noDOM anymore.

(same in the other sources, as well as in the 2.11 sources)

**Comment by sjrd** [comment]
Happy to get rid of sbt-plugin-test :tada:

**Pull request merged**